lib/chef/resource/chocolatey_source.rb

```diff
@@ -77,6 +89,26 @@ def fetch_source_element(id)
         end
       end
 
+      action :disable do
+        description "Disables a Chocolatey source."
+
+        if current_resource.source_state != true
+          converge_by("disable Chocolatey source '#{new_resource.source_name}'") do
+            shell_out!(choco_cmd("disable"))
+          end
+        end
+      end
+
+      action :enable do
+        description "Enables a Chocolatey source."
+
+        if current_resource.source_state == true
```

Comment:
this has the bug that someone who does:

```ruby
chocolatey_source "whatever" do
   action :enable
end
```

and then changes that to:

```ruby
chocolatey_source "whatever" do
  allow_self_service true
   action :enable
end
```

it'll see it is already enabled and won't fix the flag.

Reply:
This is a very good point.  If it makes any difference, the same thing will happen if you run the Chocolatey command directly as well:

![image](https://user-images.githubusercontent.com/1271146/58936534-df585480-8767-11e9-9288-9c445d67f44a.png)

This doesn't result in any changes to the allow_self_service property.

@lamont-granquist is there a strategy for how this should be handled?  Without creating a new resource, specifically for enabling/disabling a source, I don't see a path forward.  Is there a way to limit properties of a resource to only apply to certain actions?  This would then error if a property is set on an action that it wasn't intended?